File: app/serena_mcsv/mcsv-emotion/src/main/java/psytobetter/emotion/mcsv_emotion/repository/EmotionRepository.java
package psytobetter.emotion.mcsv_emotion.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import psytobetter.emotion.mcsv_emotion.model.Emotion;

import java.util.List;
import java.util.Optional;

public interface EmotionRepository extends JpaRepository<Emotion, Long> {

    Optional<Emotion> findByName(String name);

    List<Emotion> findAllByOrderByNameAsc();
}

File: app/src/main/java/com/duocuc/serena/repository/EmotionRepository.kt
package com.duocuc.serena.repository

import com.duocuc.serena.data.simpleData.emotionsList
import com.duocuc.serena.model.Emotion

class EmotionRepository {

    fun getAllEmotions(): List<Emotion> {
        return emotionsList
    }

    fun getEmotionById(id: Int): Emotion? {
        return emotionsList.find { it.id == id }
    }
}
